src/core/services/compiler/compiler.js

```diff
@@ -248,13 +266,8 @@ MdCompilerService.prototype._compileElement = function(locals, element, options)
         $element: element
       });
 
-      var invokeCtrl = self.$controller(options.controller, injectLocals, true, options.controllerAs);
-
-      if (options.bindToController) {
-        angular.extend(invokeCtrl.instance, locals);
-      }
-
-      var ctrl = invokeCtrl();
+      // Create the specified controller instance.
+      var ctrl = self._createController(options, injectLocals, locals);
```

Comment:
Currently, `injectLocals` and `locals` are the same object. I assume this a bug, but not directly related to (or introduced by) this PR.